src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py

```diff
@@ -206,1004 +209,157 @@ def create_dir(directory):
-                            if subfolder in subsubelem_dict["href"]:
-                                path_end = subsubelem_dict["href"].split(subfolder)[-1]
-                                href_corrected = f"{subfolder}{path_end}"
-                                subsubelem_dict["href"] = href_corrected
-                elem_dict.update(subsubelem_dict)
-                if subsubelem.tag == "checksum":
-                    elem_dict.update({"checksum": subsubelem.text})
-        files_list.append(elem_dict)
-    return files_list
+def normalize_time(datetime_str: str):
+    """Unifies the different ISO formats into 'YYYY-MM-DDTHH:MM:SS'
 
+    :param datetime_str: Datetime in ISO format
+    :type datetime_str: str
 
-def get_checksum(filename, hash_function="md5"):
-    """Generate checksum for file based on hash function (MD5 or SHA256).
-    Source: https://onestopdataanalysis.com/checksum/
+    :return: Datetime converted to 'YYYY-MM-DDTHH:MM:SS'
+    :rtype: str
     """
-    hash_function = hash_function.lower()
-
-    with open(filename, "rb") as f:
-        file_bytes = f.read()  # read file as bytes
-        if hash_function == "md5":
-            readable_hash = hashlib.md5(file_bytes).hexdigest()
-        elif hash_function == "sha256":
-            readable_hash = hashlib.sha256(file_bytes).hexdigest()
-        elif hash_function == "sha3-256":
-            readable_hash = hashlib.sha3_256(file_bytes).hexdigest()
-        else:
-            gs.fatal(
-                _(
-                    "{} is an invalid hash function. " "Please Enter MD5 or SHA256"
-                ).format(hash_function)
-            )
-
-    return readable_hash
-
-
-def download_gcs_file(url, destination, checksum_function, checksum):
-    """Downloads a single file from GCS and performs checksumming."""
-    # if file exists, check if checksum is ok, download again otherwise
-    if os.path.isfile(destination):
-        sum_existing = get_checksum(destination, checksum_function)
-        if sum_existing == checksum:
-            return 0
-    try:
-        r_file = requests.get(url, allow_redirects=True)
-        with open(destination, "wb") as dest:
-            dest.write(r_file.content)
-        sum_dl = get_checksum(destination, checksum_function)
-        if sum_dl.lower() != checksum.lower():
-            gs.verbose(_("Checksumming not successful for {}").format(destination))
-            return 2
-        return 0
-
-    except Exception as e:
-        gs.verbose(_("There was a problem downloading {}").format(url))
-        return 1
+    normalized_datetime = datetime.fromisoformat(datetime_str)
+    if normalized_datetime.tzinfo is None:
+        return normalized_datetime.strftime("%Y-%m-%dT%H:%M:%S")
+    return normalized_datetime.astimezone(timezone.utc).strftime("%Y-%m-%dT%H:%M:%S")
+
+
+PRODUCTTYPE_MAP = {
+    "S2MSI1C": "S2_MSI_L1C",
+    "S2MSI2A": "S2_MSI_L2A",
+    "OCN": "S1_SAR_OCN",
+    "GRD": "S1_SAR_GRD",
```

Comment:
GRD data are now provided in SAFE format with internal COG, see: https://github.com/CS-SI/eodag/pull/1277

This illutrates how hardcoding product types limits `i.sentinel.download` compared to `i.eodag`.

You should add a note in the manual that i.eodag is used under the hood and offers more comprehensive download possibilities...

Reply:
> GRD data are now provided in SAFE format with internal COG, see: [CS-SI/eodag#1277](https://github.com/CS-SI/eodag/pull/1277)

Cool, shall I add it or wait till it is officially released? 

> This illutrates how hardcoding product types limits `i.sentinel.download` compared to `i.eodag`.

Agree I think in most cases it is almost always better to use i.eodag directly, for more flexibility. Except for pre-existing code I guess.

> You should add a note in the manual that i.eodag is used under the hood and offers more comprehensive download possibilities...
 
Will do that.

Reply:
> > GRD data are now provided in SAFE format with internal COG, see: [CS-SI/eodag#1277](https://github.com/CS-SI/eodag/pull/1277)
> 
> Cool, shall I add it or wait till it is officially released?

there is a pre-release out. So I guess it is OK to add it...